File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * Created by IntelliJ IDEA.
 *
 * @Author zhangxl98
 * @Date 4/29/19 4:48 PM
 * @OS Ubuntu 18.04 LTS
 * @Device DELL-Inspiron-15-7559
 * @Modified By
 * @Version V1.0.0
 * @Description 品牌 Controller
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /**
     * 返回全部列表
     * <pre>createTime:
     * 4/29/19 3:55 PM</pre>
     *
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    /**
     * 返回分页列表
     * <pre>createTime:
     * 4/30/19 11:47 AM</pre>
     *
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page, int rows) {
        return brandService.findPage(page, rows);
    }

    /**
     * 增加品牌
     * <pre>createTime:
     * 5/5/19 10:13 AM</pre>
     *
     * @param brand
     * @return
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            // 增加成功
            return new Result(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            // 增加失败
            return new Result(false, "增加失败");
        }
    }

    /**
     * 根据 id 获取实体类
     * <pre>createTime:
     * 5/5/19 11:09 AM</pre>
     *
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }

    /**
     * 修改品牌
     * <pre>createTime:
     * 5/5/19 11:20 AM</pre>
     *
     * @param brand
     * @return
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand) {
        try {
            brandService.update(brand);
            // 修改成功
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            // 修改失败
            return new Result(false, "修改失败");
        }
    }

    /**
     * 批量删除品牌
     * <pre>createTime:
     * 5/5/19 12:36 PM</pre>
     *
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
    public Result delete(long[] ids) {
        try {
            brandService.delete(ids);
            // 删除成功
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            // 删除失败
            return new Result(false, "删除失败");
        }
    }

    /**
     * 按条件搜索并分页
     * <pre>createTime:
     * 5/6/19 10:22 AM</pre>
     *
     * @param brand
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand, int page, int rows) {
        return brandService.findPage(brand, page, rows);
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function($scope, $controller, brandService) {
  // 继承
  $controller('baseController', {
    $scope: $scope
  })

  // 读取列表数据到表单中
  $scope.findAll = () => {
    brandService.findAll().success(
      (response) => {
        $scope.list = response
      })
  }

  //分页
  $scope.findPage = (page, rows) => {
    brandService.findPage(page, rows).success(
      (response) => {
        $scope.list = response.rows
        //更新总记录数
        $scope.paginationConf.totalItems = response.total
      }
    )
  }

  // 增加或修改品牌
  $scope.save = () => {

    // 判断是增加还是修改
    let object = null
    if ($scope.entity.id == null) {
      // 增加品牌
      object = brandService.add($scope.entity)
    } else {
      // 修改品牌
      object = brandService.update($scope.entity)
    }

    object.success(
      (response) => {
        if (response.success) {
          // 成功，显示分页列表
          $scope.reloadList()
        } else {
          // 失败，弹出提示信息
          alert(response.message)
        }
      }
    )
  }

  // 根据 id 查询实体类
  $scope.findOne = (id) => {
    brandService.findOne(id).success(
      (response) => {
        $scope.entity = response
      }
    )
  }

  // 删除品牌

  // 批量删除
  $scope.delete = () => {
    brandService.delete($scope.selectIds).success(
      (response) => {
        if (response.success) {
          // 成功，显示分页列表
          $scope.reloadList()
        } else {
          // 失败，弹出提示信息
          alert(response.message)
        }
        $scope.selectIds = []
      }
    )
  }

  // 搜索
  // 定义搜索对象
  $scope.searchEntity = {}
  //条件查询
  $scope.search = (page, rows) => {
    brandService.search(page, rows, $scope.searchEntity).success(
      (response) => {
        //总记录数
        $scope.paginationConf.totalItems = response.total
        //给列表变量赋值
        $scope.list = response.rows
      }
    )
  }
  // 刷新列表
  $scope.reloadList = () => {
    $scope.search($scope.paginationConf.currentPage, $scope.paginationConf.itemsPerPage)
  }

})
